packages/editor/src/components/preview-dropdown/index.js

```diff
@@ -62,6 +63,54 @@ export default function PreviewDropdown( { forceIsAutosaveable, disabled } ) {
 		desktop,
 	};
 
+	/**
+	 * The choices for the device type.
+	 *
+	 * @type {Array}
+	 */
+	const choices = [
+		{
+			value: 'Desktop',
+			label: __( 'Desktop' ),
+			icon: desktop,
+		},
+		{
+			value: 'Tablet',
+			label: __( 'Tablet' ),
+			icon: tablet,
+		},
+		{
+			value: 'Mobile',
+			label: __( 'Mobile' ),
+			icon: mobile,
+		},
+	];
+
+	/**
+	 * The selected choice.
+	 *
+	 * @type {Object}
+	 */
+	let selectedChoice = choices.find(
+		( choice ) => choice.value === deviceType
+	);
+
+	/**
+	 * If no selected choice is found, default to the first
+	 */
+	if ( ! selectedChoice ) {
+		selectedChoice = choices[ 0 ];
+	}
+
+	/**
+	 * Handles the selection of a device type.
+	 *
+	 * @param {string} value The device type.
+	 */
+	const onSelect = ( value ) => {
+		setDeviceType( value );
+	};
```

Comment:
The changes so far look good. Right now voiceover announces that the menu item is ticked on focus which is a big improvement over what's in trunk:
![Screenshot 2024-07-29 at 12 05 12 PM](https://github.com/user-attachments/assets/944508cc-57b7-4030-ad0b-ab0728064828)

But when changing the option there's no announcement.

So another good accessibility improvement would be to add a spoken message when an option is selected (using the `speak` function from the `a11y` package). This would let screenreader users know that something happened when they change the option.

There's an example here of how the visual/code editor mode option does this:
https://github.com/WordPress/gutenberg/blob/27fbbd8e9b57495339a077ef0c8ba67a18392699/packages/editor/src/store/actions.js#L810

Something similar could be done for the device previews, either as part of this `onSelect` function or within the `setDeviceType` action.

Reply:
@talldan added `speak` on the device selection.